### ISSUE TITLE
The do action requires a RVM project file

### ISSUE THREAD

**Issue by jacob-carlborg** [rvm/rvm#3746]
If I use the `do` action in a directory which does not contain a RVM project file it fails. Example:

```
rvm / do ls
```

Then I get this error:

```
Could not determine which Ruby to use; / should contain .rvmrc or .versions.conf or 
.ruby-version or .rbfu-version or .rbenv-version, or an appropriate line in Gemfile.
( see: 'rvm usage' )
```

Is there a reason why the `do` action cannot fallback to the default or system Ruby?

**Comment by pkuczynski** [comment]
Why would you like to use rvm in this context?

**Comment by jacob-carlborg** [comment]
This come up in a discussion of implementing a function that returns the path to an executable that is written in Ruby, which would be used in TextMate to run that executable. What we need is a way to run that executable in the exact same way as if it was running from the shell. If one uses RVM it needs to select the correct Ruby to run the executable with. Normally if one runs the executable from the shell it would use the project specific Ruby, if a project file is present, the default configured Ruby with RVM, or fall back to the system Ruby.

**Comment by pkuczynski** [comment]
So you want to emulate the situation where user enters a folder and rvm reconfigures the current ruby based on the project files stored in this folder, correct?

**Comment by jacob-carlborg** [comment]
Yes. But not all projects might be using a project file. In that case, when using the shell, RVM falls back to the default Ruby or the system Ruby.

**Comment by pkuczynski** [comment]
Why don't you simply run `cd $DIR; $EXECUTABLE`. This would prepend running executable with entering the right folder

**Comment by jacob-carlborg** [comment]
Hmm, it would need to load RVM first, but that might work. But I thought this was the whole point of `rvm do`.

**Comment by noniq** [comment]
Let me give some more context: The idea is for TextMate’s Ruby bundle to provide an API for the task of “locating a Ruby executable” It is meant to be used by other bundles like RSpec or RuboCop that need to run some kind of Ruby executable that might be project-depended (different projects might use different RSpec versions etc.)

We don’t want to make any assumptions on how the executable is used, so we just return an array that can be passed to `Kernel#system` and similar functions. And if the executable is not installed at all, we return `nil` to allow the bundle to display an appropriate error message, use some kind of fallback strategy or whatever:

```ruby
# RuboCop is part of the project’s Gemfile and has a binstub:
Executable.find("rubocop") # => ["bin/rubocop"]

# RuboCop is part of the project’s Gemfile, no binstub:
Executable.find("rubocop") # => ["bundle", "exec", "rubocop"]

# RuboCop is installed globally and found in the search path.
Executable.find("rubocop") # => ["rubocop"]

# etc.

# RuboCop is not installed at all.
Executable.find("rubocop") # => nil
```

If an executable is installed globally, we use `which -s …` to check for its existence in the search path. This obviously does not work with rbenv or rvm. For rbenv, we use `rbenv which …` instead, and for rvm `rvm . do which …`. But the latter fails if the project uses the default Ruby version.
